File: LeetCode/JavaScript/Easy/climbingStairs.js
// Manuel Ortiz at 2022
// Extracted from: https://leetcode.com/problems/climbing-stairs/

const climbStairs = (n) => {
    if (n <= 2) return n;
    const memo = [];
    memo[0] = 1 
    memo[1] = 2 
    
    for (let i = 2; i < n; i++) memo[i] = memo[i-1] + memo[i-2];

    return memo[n-1];
        
};

File: LeetCode/Python/Easy/climbingStairs.py
# Manuel Ortiz at 2024
# Extracted from: https://leetcode.com/problems/climbing-stairs

class Solution:
    def climbStairsHelper(self, n, memo):
        if n<0:
            return 0
        
        if memo[n] == 0:
            memo[n] = self.climbStairsHelper(n-1, memo) + self.climbStairsHelper(n-2, memo)
    
        return memo[n]

    def climbStairs(self, n: int) -> int:
        memo = [0 for x in range(n+1)]
        memo[0] = 1 # Base Case
        return self.climbStairsHelper(n,memo)